numpy/lib/arraypad.py

```diff
@@ -34,31 +32,29 @@ def _arange_ndarray(arr, shape, axis, reverse=False):
 
     Returns
     -------
-    padarr : ndarray
-        Output array sized to pad `arr` along `axis`, with linear range from
-        1 to `shape[axis]` along specified `axis`.
+    arr : ndarray
+        Output array with `shape` that in- or decrements along the given
+        `axis`.
 
     Notes
     -----
     The range is deliberately 1-indexed for this specific use case. Think of
     this algorithm as broadcasting `np.arange` to a single `axis` of an
     arbitrarily shaped ndarray.
-
     """
-    initshape = tuple(1 if i != axis else shape[axis]
-                      for (i, x) in enumerate(arr.shape))
+    init_shape = (1,) * axis + (shape[axis],) + (1,) * (len(shape) - axis - 1)
     if not reverse:
-        padarr = np.arange(1, shape[axis] + 1)
+        arr = np.arange(1, shape[axis] + 1)
     else:
-        padarr = np.arange(shape[axis], 0, -1)
-    padarr = padarr.reshape(initshape)
+        arr = np.arange(shape[axis], 0, -1)
+    arr = arr.reshape(init_shape)
     for i, dim in enumerate(shape):
-        if padarr.shape[i] != dim:
-            padarr = padarr.repeat(dim, axis=i)
-    return padarr
+        if arr.shape[i] != dim:
+            arr = arr.repeat(dim, axis=i)
```

Comment:
I think `broadcast_to` would be faster here

Reply:
You are correct. Actually we don't even need to do this. We only need to ensure that the resulting array is broadcastable to the final shape then NumPy handles that automatically. This allows us to simplify some parts of the algorithm and testing on my local machine hints that this is slightly faster too.

Note: I'll do extensive profiling and optimization later, once the first draft is ready.